spec/models/adoption_application_spec.rb

```diff
@@ -0,0 +1,40 @@
+require "rails_helper"
+
+RSpec.describe AdoptionApplication do
+
+  describe "relationships" do
+    it {should have_many :pet_applications}
+    it {should have_many(:pets).through(:pet_applications)}
+  end
+
+  describe "validations" do
+    it { should validate_presence_of :name}
+    it { should validate_presence_of :address}
+    it { should validate_presence_of :city}
+    it { should validate_presence_of :state}
+    it { should validate_presence_of :zip}
+    it { should validate_presence_of :phone_number}
+    it { should validate_presence_of :description}
+  end
+
+  describe "class methods" do
+    it "#pets" do
+      shelter = Shelter.create(name: "Happy Puppies", address: "55 Street St", city: "Danger Mountain", state: "UT", zip: "80304")
+
+      pet_1 = Pet.create(image: "image.jpeg", name: "Kunga", approximate_age: "1", sex: "male", shelter_id: shelter.id)
+      pet_2 = Pet.create(image: "image.jpeg", name: "Honey Pie", approximate_age: "11", sex: "female", shelter_id: shelter.id)
+
+      application_1 = AdoptionApplication.create(name: "Stella", address: "street", city: "City", state: "ST", zip: "34567", phone_number: "545-567-7643", description: "I'm awesome")
+      application_2 = AdoptionApplication.create(name: "Stella", address: "street", city: "City", state: "ST", zip: "34567", phone_number: "545-567-7643", description: "I'm still awesome")
+
+      PetApplication.create(pet_id: pet_1.id, adoption_application_id: application_1.id)
+      PetApplication.create(pet_id: pet_2.id, adoption_application_id: application_1.id)
+      PetApplication.create(pet_id: pet_1.id, adoption_application_id: application_2.id)
+      PetApplication.create(pet_id: pet_2.id, adoption_application_id: application_2.id)
```

Comment:
by creating the PetApplication objects like this I am able to verify that an application can have many pets and a pet can have many applications, even though the new application form isn't able to handle multiple pets.